src/main/scala/java/time/Instant.scala

```diff
@@ -0,0 +1,267 @@
+    case HOURS     => plusSeconds(amount * SECONDS_IN_HOUR)
+    case HALF_DAYS => plusSeconds((amount * SECONDS_IN_DAY) / 2)
+    case DAYS      => plusSeconds(amount * SECONDS_IN_DAY)
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+
+    case _ => unit.addTo(this, amount)
+  }
+
+  def plusSeconds(secs: Long): Instant = plus(secs, 0)
+
+  def plusMillis(millis: Long): Instant = plusNanos(millis * NANOS_IN_MILLI)
+
+  def plusNanos(nans: Long): Instant = plus(0, nans)
+
+  private def plus(secs: Long, nans: Long): Instant = {
+    if (secs == 0 && nans == 0) this
+    else {
+      val secondsFromNanos = Math.floorDiv(nans, NANOS_IN_SECOND)
+      val remainingNanos = Math.floorMod(nans, NANOS_IN_SECOND)
+      val additionalSecs = Math.addExact(secs, secondsFromNanos)
+      ofEpochSecond(
+        Math.addExact(seconds, additionalSecs),
+        Math.addExact(nanos, remainingNanos)
+      )
+    }
+  }
+
+  override def minus(amount: Long, unit: TemporalUnit): Instant = {
+    if (amount == Long.MinValue) plus(Long.MaxValue, unit).plus(1, unit)
+    else plus(-amount, unit)
+  }
+
+  def minusSeconds(secs: Long): Instant = minus(secs, SECONDS)
+
+  def minusMillis(millis: Long): Instant = minus(millis, MILLIS)
+
+  def minusNanos(nans: Long): Instant = minus(nans, NANOS)
+
+  // Not implemented
+  // def query[R](query: TemporalQuery[R]): R
+
+  def adjustInto(temporal: Temporal): Temporal =
+    temporal.`with`(INSTANT_SECONDS, seconds).`with`(NANO_OF_SECOND, nanos)
+
+  def until(end: Temporal, unit: TemporalUnit): Long = {
+    val endInstant = from(end)
+    val nanosDiff = endInstant.totalNanos - totalNanos
+    unit match {
+      case NANOS     => nanosDiff
+      case MICROS    => nanosDiff / NANOS_IN_MICRO
+      case MILLIS    => nanosDiff / NANOS_IN_MILLI
+      case SECONDS   => nanosDiff / NANOS_IN_SECOND
+      case MINUTES   => nanosDiff / NANOS_IN_MINUTE
+      case HOURS     => nanosDiff / NANOS_IN_HOUR
+      case HALF_DAYS => nanosDiff / (NANOS_IN_HOUR * 12)
+      case DAYS      => nanosDiff / NANOS_IN_DAY
+
+      case _: ChronoUnit =>
+        throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+
+      case _ => unit.between(this, end)
+    }
+  }
+
+  // Not implemented
+  // def atOffset(offset: ZoneOffset): OffsetDateTime
+
+  // Not implemented
+  // def atZone(zone: ZoneId): ZonedDateTime
+
+  def toEpochMilli(): Long = Math.floorDiv(totalNanos, NANOS_IN_MILLI)
+
+  def compareTo(that: Instant): Int =
+    totalNanos.compareTo(that.totalNanos)
+
+  def isAfter(that: Instant): Boolean = compareTo(that) > 0
+
+  def isBefore(that: Instant): Boolean = compareTo(that) < 0
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Instant => totalNanos == that.totalNanos
+    case _             => false
+  }
+
+  override def hashCode(): Int = totalNanos.hashCode()
```

Comment:
Experimenting a bit on the JVM, I found that their implementation is `(seconds + 51*nanos).hashCode`.